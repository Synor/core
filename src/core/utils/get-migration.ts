import { SynorMigration } from '../migration'

type MigrationSource = import('../migration').MigrationSource
type MigrationType = import('../migration').MigrationType
type MigrationVersion = import('../migration').MigrationVersion
type SourceEngine = import('../source').SourceEngine

export async function getMigration(
  source: SourceEngine,
  version: MigrationVersion,
  type: MigrationType
): Promise<MigrationSource | null> {
  const info = await source.get(version, type)

  if (!info) {
    return null
  }

  const content = await source.read(info)

  const migration = SynorMigration(info, content)

  return migration
}
