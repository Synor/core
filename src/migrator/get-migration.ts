import { SynorMigration } from '../migration'

type MigrationSource = import('../migration').MigrationSource
type MigrationType = import('../migration').MigrationType
type SourceEngine = import('../source').SourceEngine

export async function getMigration(
  source: SourceEngine,
  version: string,
  type: MigrationType
): Promise<MigrationSource | null> {
  const info = await source.get(version, type)

  if (!info) {
    return null
  }

  const content = await source.read(info)

  const migration = SynorMigration(info, content)

  return migration
}
