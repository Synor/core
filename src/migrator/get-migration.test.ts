import { getMigration } from './get-migration'

const source = {
  get: jest.fn(),
  read: jest.fn()
}

describe('migrator:getMigration', () => {
  test('returns null (if not available)', async () => {
    source.get.mockImplementationOnce(() => Promise.resolve(null))

    await expect(getMigration(source as any, '99', 'do')).resolves.toBeNull()
    expect(source.get).toBeCalledWith('99', 'do')

    source.get.mockReset()
  })

  test('returns data (if available)', async () => {
    const info = { version: '01', type: 'do', title: '', filename: '' }

    source.get.mockImplementationOnce(() => Promise.resolve(info))
    source.read.mockImplementationOnce(() => Promise.resolve({ body: '' }))

    await expect(
      getMigration(source as any, '01', 'do')
    ).resolves.toMatchSnapshot()
    expect(source.get).toBeCalledWith('01', 'do')
    expect(source.read).toBeCalledWith(info)

    source.get.mockReset()
    source.read.mockReset()
  })
})
