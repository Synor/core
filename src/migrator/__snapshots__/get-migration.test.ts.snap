// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrator:getMigration returns data (if available) 1`] = `
Object {
  "body": "",
  "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
  "title": "",
  "type": "do",
  "version": "01",
}
`;
